src/Discord.Net.WebSocket/Entities/Interaction/Context Menu Commands/Message Commands/SocketApplicationMessageCommandData.cs

```diff
@@ -0,0 +1,140 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Model = Discord.API.ApplicationCommandInteractionData;
+
+namespace Discord.WebSocket
+{
+    /// <summary>
+    ///     Represents the data tied with the <see cref="SocketApplicationMessageCommand"/> interaction.
+    /// </summary>
+    public class SocketApplicationMessageCommandData : SocketEntity<ulong>, IApplicationCommandInteractionData
+    {
+        /// <inheritdoc/>
+        public string Name { get; private set; }
+        public SocketMessage Message {  get; private set; }
+
+        internal Dictionary<ulong, SocketGuildUser> guildMembers { get; private set; }
```

Comment:
For resolving types there should be a base `SocketApplicationCommandData` class that handles resolvable types